test/framework/awsiam.go

```diff
@@ -0,0 +1,90 @@
+package framework
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	eksdv1alpha1 "github.com/aws/eks-distro-build-tooling/release/api/v1alpha1"
+
+	"github.com/aws/eks-anywhere/internal/pkg/api"
+	"github.com/aws/eks-anywhere/internal/pkg/awsiam"
+	"github.com/aws/eks-anywhere/pkg/cluster"
+	"github.com/aws/eks-anywhere/pkg/version"
+)
+
+const (
+	AWSIamRoleArn = "T_AWS_IAM_ROLE_ARN"
+	AWSIamUserArn = "T_AWS_IAM_USER_ARN"
+)
+
+var awsIamRequiredEnvVars = []string{
+	AWSIamRoleArn,
+}
+
+func RequiredAWSIamEnvVars() []string {
+	return awsIamRequiredEnvVars
+}
+
+func WithAWSIam() E2ETestOpt {
+	return func(e *E2ETest) {
+		checkRequiredEnvVars(e.T, awsIamRequiredEnvVars)
+		e.AWSIamConfig = api.NewAWSIamConfig(defaultClusterName,
+			api.WithAWSIamAWSRegion("us-west-1"),
+			api.WithAWSIamClusterID(defaultClusterName),
+			api.WithAWSIamPartition("aws"),
+			api.WithAWSIamBackendMode("EKSConfigMap"),
+			api.AddAWSIamMapRoles(api.WithAWSIamRole(withArnFromEnv(AWSIamRoleArn), "kubernetes-admin", []string{"system:masters"})),
+		)
+		e.clusterFillers = append(e.clusterFillers,
+			api.WithAWSIamIdentityProviderRef(defaultClusterName),
+		)
+	}
+}
+
+func withArnFromEnv(envVar string) string {
```

Comment:
didn't we have something already to do this?

Reply:
I did check this and most of them return a particular "[option](https://github.com/aws/eks-anywhere/blob/main/internal/pkg/api/gitops.go#L72)" or [Type](https://github.com/aws/eks-anywhere/blob/main/internal/pkg/api/vsphere.go#L290). 
I just had this method for getting the value.
Is there something I am missing out?